marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/ConstraintsSpec.hs

```diff
@@ -0,0 +1,178 @@
+{-# LANGUAGE DataKinds #-}
+
+module Language.Marlowe.Runtime.Transaction.ConstraintsSpec
+  ( spec
+  ) where
+
+import qualified Cardano.Api as C
+import qualified Cardano.Api.Shelley as C
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.Map as Map
+import Data.Monoid (First(..), getFirst)
+import Data.Ratio ((%))
+import qualified Data.Set as Set
+import Gen.Cardano.Api.Typed
+import Language.Marlowe.Runtime.Cardano.Api (fromCardanoAddressAny, fromCardanoAddressInEra)
+import qualified Language.Marlowe.Runtime.ChainSync.Api as Chain
+import qualified Language.Marlowe.Runtime.Core.Api as Core.Api
+import Language.Marlowe.Runtime.Transaction.Constraints (TxConstraints(..))
+import qualified Language.Marlowe.Runtime.Transaction.Constraints as Constraints
+import qualified Language.Marlowe.Runtime.Transaction.Constraints as TxConstraints
+import Test.Hspec (Spec, expectationFailure, shouldBe)
+import qualified Test.Hspec as Hspec
+import qualified Test.Hspec.QuickCheck as Hspec.QuickCheck
+import qualified Test.QuickCheck as QuickCheck
+import Test.QuickCheck.Hedgehog (hedgehog)
+
+byteStringGen :: QuickCheck.Gen ByteString
+byteStringGen = BS.pack <$> QuickCheck.arbitrary
+
+emptyTxConstraints :: TxConstraints 'Core.Api.V1
+emptyTxConstraints = TxConstraints
+  { marloweInputConstraints = TxConstraints.MarloweInputConstraintsNone
+  , payoutInputConstraints = Set.empty
+  , roleTokenConstraints = TxConstraints.SpendRoleTokens Set.empty
+  , payToAddresses = Map.empty
+  , payToRoles = Map.empty
+  , marloweOutputConstraints = TxConstraints.MarloweOutputConstraintsNone
+  , merkleizedContinuationsConstraints = mempty
+  , signatureConstraints = Set.empty
+  , metadataConstraints = Map.empty
+  }
+
+spec :: Spec
+spec = do
+  Hspec.QuickCheck.prop "mustSpendToken" do
+    policyId <- Chain.PolicyId <$> byteStringGen
+    tokenName <- Chain.TokenName <$> byteStringGen
+
+    let
+      assetId :: Chain.AssetId
+      assetId = Chain.AssetId policyId tokenName
+
+      actual :: Constraints.TxConstraints 'Core.Api.V1
+      actual = Constraints.mustSpendRoleToken assetId
+
+      expected :: Constraints.TxConstraints 'Core.Api.V1
+      expected = emptyTxConstraints
+        { roleTokenConstraints = Constraints.SpendRoleTokens $ Set.singleton assetId }
+
+    pure $ actual `shouldBe` expected
```

Comment:
Why is this test here?

Reply:
Ah, this isn't about adjustTxForMinUtxo at all. I added it when first trying to write tests for this module and then forgot it was here. I got rid of it for now and saved the source code somewhere.